aspnet/fundamentals/file-providers/sample/src/FileProviderSample/Controllers/HomeController.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.FileProviders;
+
+namespace FileProviderSample.Controllers
+{
+    public class HomeController : Controller
+    {
+        private readonly IFileProvider _fileProvider;
+        private static string _dateFileLastChanged = "Never";
+
+        public HomeController(IFileProvider fileProvider)
```

Comment:
I'd be ok with that as long as the Startup class only registers one `IFileProvider` implementation. Currently it registers three. As a principle, we're try to avoid scenarios where a component depends on the order in which services are registered.